### PR TITLE
fix: change service initialization order #567

### PR DESCRIPTION
**Pull request by nhardman**

### CODE REVIEW

**Review by tunniclm — COMMENTED** [review]
Is there a reason to push the metrics code into the services segment, can it be done just in the initialize?

File: refresh/templates/common/Application.swift [hunk 1]
```diff
@@ -26,18 +26,19 @@ public var basePath = "<%- basepath %>"
...
-
 class ApplicationServices {
-    // Service references
+    // Initialize services
 <% appInitCode.service_variables.forEach(function(variable) { -%>
     <%- variable %>
 <% }); -%>
 
     public init(cloudEnv: CloudEnv) throws {
+<% if (appInitCode.services.length) { -%>
```
Comment (tunniclm) [inline refresh/templates/common/Application.swift]: Isn't this redundant with the check on line 28?

File: refresh/templates/common/Application.swift [hunk 2]
```diff
@@ -26,18 +26,19 @@ public var basePath = "<%- basepath %>"
 public let health = Health()
 <% } -%>
 <% if (appInitCode.services.length > 0) { -%>
-
```
Comment (tunniclm) [inline refresh/templates/common/Application.swift]: Removing the blank line here will mean the generated app won't have a blank line between the definition of `health` and the start of the `ApplicationServices` class any more; was that the intention?

File: refresh/templates/common/Application.swift (hunk 1 above)
Comment (nhardman) [reply refresh/templates/common/Application.swift]: fixed.

File: refresh/templates/common/Application.swift (hunk 2 above)
Comment (nhardman) [reply refresh/templates/common/Application.swift]: no, it was a mistake. Now fixed

**Review by tunniclm — APPROVED** [review]

**Pull request merged**